general/v0.1/registration.md

```diff
@@ -27,112 +27,101 @@ administrator. This MAY include but is not limited to the following:
 
-![A bare-bones web form to add a provider with a textarea for the registration token](../../images/add_provider.svg)
+It MUST display a fingerprint of the FASP's public key for comparison
+purposes. The fingerprint is the Base64 encoded SHA-256 hash of the
+public key.
 
-After successful submission of this data, the fediverse software MUST
-persist it and initiate an OAuth 2.0 "client credentials" flow to
-obtain an access token. This token can then be used to authenticate
-subsequent API calls.
+The fediverse server MUST present a list of FASP registration requests
+to the administrator. This list MUST be accessible via regular means,
+i.e. a navigation item in the administration area.
 
-If authorization was successful the instance MUST create and persist an
-OAuth 2.0 application representing the FASP, including a client ID
-and a secret.
+The `redirectUri` MAY lead to this list, optionally highlighting or
+expanding the registration in question. Alternatively it MAY lead to
+page that only displays the registration in question.
 
-It MUST communicate its base URL, the generated client ID and secret to
-the FASP using the client credentials API endpoint.
+For each registration request the fediverse server MUST display the
+`name` the FASP sent and the fingerprint of its public key.
 
-Example request:
+The administrator MUST be able to either accept or decline a
+registration request.
 
-OAuth 2.0 scope: `setup`
+The following is a sketch of how this may look in the abstract:
 
-```http
-POST /client_credentials
-```
+Step 1: A fediverse server admin is presented with a registration form
 
-Example request body:
+![A bare-bones sign-up form asking for email, server base URL and acceptance of terms of service](../../images/server_sign_up.svg)
 
-```json
-{
-  "baseURL": "https://fediverse-server.example.com",
-  "clientId": "kOP2Bc7oP31zM",
-  "clientSecret": "pl5z4ciwfEP194MwWziP"
-}
-```
+Step 2: Upon successful registration, the fingerprint of the public key
+and a link / button to the fediverse server is displayed
 
-The FASP MUST respond with an HTTP status code `201` (Created)  and
-persist the ID, secret and base URL it received.
+![A webpage displaying a public key fingerprint with a button to go to the fediverse server](../../images/server_sign_up_success.svg)
 
-Until this ID, secret and base URL have been communicated successfully,
-FASP MAY refuse to accept API requests and respond with the HTTP
-status code `424` (Failed Dependency) instead.
+Step 3: The fediverse servers displays FASP registration requests,
+allows to compare name and fingerprint and then to either accept or deny
+the requests.
 
-Fediverse software SHOULD catch this and retry sending the data in this
-case.
+![A list of FASP registration requests that can be expanded. The one
+expanded entry shows the fingerprint and buttons to either accept or
+deny the request.](../../images/fasp_registration_requests.svg)
```

Comment:
It may make sense to have the capabilities requested/provided be listed here, these could be either communicated via a well known file for the FASP, or by the registration request (e.g., registration request might be only registering for a single capability out of many that the FASP provides)

Reply:
Could it be that you missed the section on capabilites at the bottom of this document and the `provider_info.md` file?

Those have not changed with HTTP Message Signatures and are thus not easily visible when just looking at the diff.

Reply:
Maybe, I'll take another look when I've more energy.